activesupport/lib/active_support/core_ext/enumerable.rb

```diff
@@ -157,6 +157,21 @@ def pluck(*keys)
     end
   end
 
+  # Extract the given key from the first element in the enumerable.
+  #
+  #   [{ name: "David" }, { name: "Rafael" }, { name: "Aaron" }].pick(:name)
+  #   # => "David"
+  #
+  #   [{ id: 1, name: "David" }, { id: 2, name: "Rafael" }].pick(:id, :name)
+  #   # => [1, "David"]
+  def pick(*keys)
```

Comment:
This method works only on array of hashes, not for all enumerables.

```ruby
irb(main):012:0> [].pick
Traceback (most recent call last):
        1: from (irb):12
NoMethodError (undefined method `[]' for nil:NilClass)
irb(main):013:0> {}.pick
Traceback (most recent call last):
        1: from (irb):13
NoMethodError (undefined method `[]' for nil:NilClass)
irb(main):014:0> {}.pick(:a)
Traceback (most recent call last):
        1: from (irb):14
NoMethodError (undefined method `[]' for nil:NilClass)
irb(main):015:0> [{}].pick(:a)
=> nil
```

pluck works for those cases

```ruby
irb(main):016:0> [].pluck
=> []
irb(main):017:0> {}.pluck
=> []
irb(main):018:0> {}.pluck(:a)
=> []
irb(main):019:0> [{}].pluck(:a)
=> [nil]
```

Reply:
Good catch! I forgot to test the empty case.

The results are now consistent with `pluck(...).first` here too:

```ruby
irb(main):001:0> [].pick
=> nil
irb(main):002:0> [].pluck.first
=> nil
irb(main):003:0> {}.pick
=> nil
irb(main):004:0> {}.pluck.first
=> nil
irb(main):005:0> {}.pick(:a)
=> nil
irb(main):006:0> {}.pluck(:a).first
=> nil
irb(main):007:0> [{}].pick(:a)
=> nil
irb(main):008:0> [{}].pluck(:a).first
=> nil
```